src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp

```diff
@@ -174,23 +174,41 @@ memory::ptr ocl_engine::allocate_memory(const layout& layout, allocation_type ty
 
     check_allocatable(layout, type);
 
+    auto zero_bytes_layout = false;
```

Comment:
[random spot] I think previously we tried to avoid attempts to allocate 0 bytes on external code as it's something not expected (you can't do anything with buffer after that). Is it some special case? Maybe we can do a fix on caller code side?

Reply:
I've updated to move the alloc-1dim-n-reinterpret-to-0dim code to util code instead of modify allocate_memory() in ocl_engine directly.
Loop input node like parameter type can have {0, 0} shape for input/output and the alloc-1dim-n-reinterpret-to-0dim code is necessary for that. Or any better idea?

**(snippet from TFLite model1_ov.xml)**
<layer id="7" name="model/bidirectional/forward_lstm_1/while/Placeholder_1" type="Parameter" version="opset1">
    <data shape="0,0" element_type="f32" />
    <rt_info>
        <attribute name="fused_names" version="0" value="model/bidirectional/forward_lstm_1/while/Placeholder_1" />
        <attribute name="old_api_map_element_type" version="0" value="f16" />
    </rt_info>
    <output>
        <port id="0" precision="FP32">
            <dim>0</dim>
            <dim>0</dim>
        </port>
    </output>
</layer>